domain/consensus/processes/difficultymanager/difficultymanager.go

```diff
@@ -67,19 +67,40 @@ func (dm *difficultyManager) genesisBits() (uint32, error) {
 	return header.Bits(), nil
 }
 
-// RequiredDifficulty returns the difficulty required for some block
-func (dm *difficultyManager) UpdateDAADataAndReturnDifficultyBits(blockHash *externalapi.DomainHash) (uint32, error) {
+// StageDAADataAndReturnRequiredDifficulty calculates the DAA window, stages the DAA score and DAA added
+// blocks, and returns the required difficulty for the given block.
+// The reason this function both stages DAA data and returns the difficulty is because in order to calculate
+// both of them we need to calculate the DAA window, which is a relatively heavy operation, so reuse the block
+// window instead of recalculating it for the two purposes.
+// For cases where no staging should be happen and the caller only needs to know the difficulty RequiredDifficulty
```

Comment:
Grammar: no staging should **be** happen

I'd also rephrase "... the caller only needs to know the difficulty RequiredDifficulty should be used" to "... the caller only needs to know the difficulty - use RequiredDifficulty instead". (If you disagree at least add some separator before RequiredDifficulty)